src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/mercury/BasePageObject.java

```diff
@@ -66,4 +67,38 @@ public void waitForLoadingSpinnerToFinishReloadingPage() {
       waitForElementPresenceByBy(By.cssSelector(".loading-overlay.hidden"));
     }
   }
+
+  //TODO: Remove this and use combination od logUrl
+  public boolean isUrlPathEqualTo(String path) {
+    String currentPath = new UrlBuilder().getUrlPath(driver);
+    return currentPath.equals(path);
+  }
+
+  public CuratedMainPagePageObject openCuratedMainPage(String wikiURL, String mainPage) {
+    getUrl(wikiURL + URLsContent.WIKI_DIR + mainPage + "?cb=" + DateTime.now().getMillis());
+    PageObjectLogging
+        .log("openCuratedMainPage", "Curated main page" + mainPage + " was opened", true);
+    return new CuratedMainPagePageObject(driver);
+  }
+
+  private enum Settings {
+    TIME_OUT_IN_SEC(5),
+    CHECK_OUT_IN_MILLI_SEC(1000);
+
+    private int value;
+
+    private Settings(int value) {
+      this.value = value;
+    }
+  }
+
+  public boolean isElementVisible(WebElement element) {
+    try {
+      waitForElementVisibleByElement(element, Settings.TIME_OUT_IN_SEC.value,
+                                     Settings.CHECK_OUT_IN_MILLI_SEC.value);
+    } catch (TimeoutException e) {
```

Comment:
what about exceptions like StaleElementReferenceException and NoSuchElementExceptions?